timeboost-contract/src/events.rs

```diff
@@ -0,0 +1,324 @@
+//! Event monitoring and filtering for KeyManager contract
+use crate::bindings::keymanager::KeyManager;
+use alloy::{
+    primitives::Address,
+    providers::Provider,
+    rpc::types::{BlockNumberOrTag, Filter},
+    sol_types::SolEvent,
+};
+use anyhow::Result;
+
+pub struct KeyManagerEventMonitor<P> {
+    provider: P,
```

Comment:
I wonder if we should hold `Arc<P>` instead, given that we are cloning it to create this struct and to create `CommitteeManager`.

Reply:
thanks for this,updated here https://github.com/EspressoSystems/timeboost/pull/495/commits/7a586666a5b7ba278cfb7efcc10553ffff0d9168

Reply:
thanks, btw the variable name doesn't need `_arc`. we don't usually expose the kind of smart pointer the data resides it.